text/0374-jsii-modular.md

```diff
@@ -0,0 +1,256 @@
+code around will also require some time to be invested: there are currently
+co-dependent relationships between certain packages (`jsii-pacmak` has a local
+develooment dependency on the runtimes, and on the `jsii-calc` packages, etc...)
+that will need to be broken. Additionally, some checks currently enforce that
+all tools used to work on an assembly are on the same version line, which will
+no longer be the case.
+
+### What is the technical solution (design) of this feature?
+
+The first component of implementing this RFC is to migrate code for various
+tools and libraries to their own repository:
+
+- `@jsii/spec` and `jsii-reflect` (those can be combined as they have similar
+  purpose)
+- `jsii`
+- `jsii-pacmak`
+- `jsii-rosetta`
+- `@jsii/kernel` and `@jsii/runtime` (which could be combined into a single
+  package)
+- Each individual `@jsii/*-runtime` package (and related tests)
+
+Those new repositories could be created within the `aws` GitHub organization, or
+alternatively a new organization could be created (the `jsii` name is already in
+use as a GitHub organization).
+
+The second component of this RFC is changing the release strategy for the `jsii`
+compiler so that it follows `typescript` version lines (`major.minor`), and
+ensuring other downstream consumers of `.jsii` assemblies do not attempt to
+enforce version consistency using the `jsiiVersion` field.
+
+### Is this a breaking change?
+
+This is not a breaking change in the traditional sense: all existing code will
+continue to work with the current release lines of all packages in the
+ecosystem. However, the new versioning strategy of `jsii` is a departure from
+Semantic Versioning, which will be accompanied with a major version change (from
+`1.x` to `4.3.x` - matching the current `typescript` release line). This is to
+be noted in the release ntoes.
+
+### What alternative solutions did you consider?
+
+An alternative to de-coupled versioning is to time breaking changes in the
+*jsii* constellation with new major releases of the root consumer libraries
+(`constructs`, `aws-cdk-lib`, ...), however this forgets to account for the
+existence of other consumers of *jsii* which do not depend on these libraries,
+and may hence not have the same drive to release a new major version at that
+time. Additionally, such major releases are expected to be infrequent as they
+represent a major burden on the community, meaning our ability to release
+improvements to *jsii* packages would be severely limited.
+
+It would also be possible to unlock versions in the mono-repository, in order to
+allow each packages to include breaking changes when necessary. However, this
+makes it all too easy to inadvertently break backwards compatibility, as it is
+so tempting to introduce the fix for a breaking change in the same PR as the
+breaking change itself. Using several repositories makes this much less likely.
+
+### What are the drawbacks of this solution?
```

Comment:
What about your current PR validation (and build?) process that uses GH actions? 
Any particular challenges in that area?